### PR TITLE
📖  amp-iframe vs native iframe documentation

### PR DESCRIPTION
**Pull request by dmanek**
Half pager describing if we still need amp-iframe component or it can retired in favor or native iframes. Partial for #35147.

### CODE REVIEW

File: extensions/amp-iframe/amp-iframe-vs-native-iframe.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+amp-iframe has been an AMP component since ~2015. It serves as a wrapper around native iframes and enforces restrictions like its position in an AMP document to not degrade user experience. It also includes features like dynamic resizing. With the migration to Bento, we want to take a step back and assess if we need an amp-iframe component, given native iframes now support features like lazy loading.
+
+## Do we need amp-iframe?
+
+Yes. We cannot retire amp-iframe since there are several features that it includes that cannot be satisfied using native iframe. Some of these are discussed in detail below.
+
+### AMP Cache
+
+When an AMP document is viewed in the AMP viewer, the content is typically loaded from the AMP cache. We need to ensure that the  content is not preloaded and cached, which is not possible with native iframes.
```
Comment (dmanek) [inline extensions/amp-iframe/amp-iframe-vs-native-iframe.md:13]: @kristoferbaxter not sure if I've captured this accurately?

File: extensions/amp-iframe/amp-iframe-vs-native-iframe.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+amp-iframe has been an AMP component since ~2015. It was designed to make native iframes more secure while
+enforcing restrictions like its position outside the top 75% or 600px in an AMP document to not degrade user experience. With the migration of amp-iframe to Bento, we want to assess if we retire amp-iframe component in favor of native iframes, which now support new features like lazy loading.
+
+## So - can we retire amp-iframe?
+
+No, we still need the amp-iframe component. There are several features that cannot be implemented using native iframes, like the ones discussed below.
+
+### AMP Cache
+
+Content is preloaded and cached by the AMP cache and displayed in the AMP viewer when an AMP document is requested. With native iframes, we cannot ensure that the content is not preloaded or cached.
```
Comment (dmanek) [inline extensions/amp-iframe/amp-iframe-vs-native-iframe.md:14]: @kristoferbaxter not sure if I've captured this accurately?

File: extensions/amp-iframe/amp-iframe-vs-native-iframe.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# amp-iframe vs native iframe
+
+## Background
+
+amp-iframe has been an AMP component since ~2015. It was designed to make native iframes more secure while
+enforcing restrictions like its position outside the top 75% or 600px in an AMP document to not degrade user experience. With the migration of amp-iframe to Bento, we want to assess if we retire amp-iframe component in favor of native iframes, which now support new features like lazy loading.
```
Comment (kristoferbaxter) [inline extensions/amp-iframe/amp-iframe-vs-native-iframe.md:6]: Nit: extra carriage return?

Also, there are multiple options here. Could we allow `iframe` in addition to `amp-iframe`?

File: extensions/amp-iframe/amp-iframe-vs-native-iframe.md (hunk 2 above)
Comment (kristoferbaxter) [reply extensions/amp-iframe/amp-iframe-vs-native-iframe.md:14]: Can you elaborate on "we cannot ensure that the content is not preloaded or cached."

Are there strategies for making this work with lazy loaded iframes that could be broadly adopted? What are the downsides of these approaches?

File: extensions/amp-iframe/amp-iframe-vs-native-iframe.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+No, we still need the amp-iframe component. There are several features that cannot be implemented using native iframes, like the ones discussed below.
+
+### AMP Cache
+
+Content is preloaded and cached by the AMP cache and displayed in the AMP viewer when an AMP document is requested. With native iframes, we cannot ensure that the content is not preloaded or cached.
+
+### Supporting Dynamic Resizing, Viewability and Consent Data
+
+Currently, amp-iframe supports features like [dynamic resizing](https://amp.dev/documentation/components/amp-iframe/#iframe-resizing), [viewability](https://amp.dev/documentation/components/amp-iframe/#iframe-viewability) and [sending consent data](https://amp.dev/documentation/components/amp-iframe/#iframe-&-consent-data) to the parent frame by sending a `postMessage` to the parent window. These require including custom JS which is not trivial with native iframes.
```
Comment (kristoferbaxter) [inline extensions/amp-iframe/amp-iframe-vs-native-iframe.md:18]: Could this message protocol be standardized? Are there emerging web specifications that cover the concern we could monitor and re-evaluate?

File: extensions/amp-iframe/amp-iframe-vs-native-iframe.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+### Cross browser support
+
+Lazy loading using `loading="lazy"` attribute is currently only supported in Chromium based browsers (https://caniuse.com/loading-lazy-attr). And the thresholds to lazy load resources may differ among browsers. Chrome, for example, takes into account connection type, whether Lite mode is enabled, etc. The differences between browser implementations may lead to inconsistent behavior if we choose to support native iframes with lazy loading.
```
Comment (kristoferbaxter) [inline extensions/amp-iframe/amp-iframe-vs-native-iframe.md:22]: This is a pretty big point: But important to call out. We'd lose the lazy nature of loading outside of Chrome without an approach that polyfills.

**Review by kristoferbaxter — APPROVED** [review]
Left some comments. Overall good, but could be explained more thoroughly so we can re-evaluate over time.

**Pull request merged**